geonode/documents/views.py

```diff
@@ -336,17 +346,13 @@ def form_valid(self, form):
         If the form is valid, save the associated model.
         """
         file = form.cleaned_data.get('doc_file')
-        if file:
-            # If the files exists, we will replace the file present via the storage_manager
-            from pathlib import Path
-            path = str(Path(self.object.files[0]).parent.absolute())
-            old_file_name, _ = os.path.splitext(os.path.basename(self.object.files[0]))
-            _, ext = os.path.splitext(file.name)
-            filepath = storage_manager.save(f"{path}/{old_file_name}{ext}", file)
-            self.object.files = [storage_manager.path(filepath)]
-
-        self.object = form.save()
+
+        self.object = resource_manager.replace(self.object, file, self.request.user)
+
+        #create_document_thumbnail(object_id=self.object.resourcebase_ptr.id)
```

Comment:
Sholdn't we call "resource_manager.set_thumbnail" here?

Reply:
Uhm no, because from the code, the `resource_manager.set_thumbnail` will just call the `_concrete_resource_manager.set_thumbnail` which excludes the `Documents`.
In this case, the `resource_manager.set_thumbnail` is called which will take care of, also update the thumbnail of the document (by calling inside the `resource_manager.update` )